staging/src/k8s.io/client-go/tools/clientcmd/api/helpers.go

```diff
@@ -189,3 +188,79 @@ func MakeAbs(path, base string) (string, error) {
 	}
 	return filepath.Join(base, path), nil
 }
+
+// RedactSecrets replaces any sensitive values with REDACTED
+func RedactSecrets(config *Config) error {
+	if err := redactSecrets(reflect.ValueOf(config), false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func redactSecrets(curr reflect.Value, redact bool) error {
```

Comment:
As far as I can see, `datapolicy` is set only for 3 fields https://github.com/kubernetes/kubernetes/blob/7e079f5144474cae05802771f604df2b748d781f/staging/src/k8s.io/client-go/tools/clientcmd/api/types.go#L117-L120 and `Config` struct is not something very often changing.

Would it be better to redact just these ^ fields statically instead this dynamic recursive function?.

I wonder other reviewers opinion about this.